src/api/routes/offer.js

```diff
@@ -28,6 +29,25 @@ module.exports = (app) => {
         }
     });
 
+    /**
+     * Gets an offer from the database with its id
+    */
+    router.get("/:offer", validators.get, async (req, res, next) => {
+        try {
+            const offer = await (new OfferService()).getOfferById(req.params.offer);
+            if (offer !== null) {
+                return res.json(offer);
+            } else {
+                return res.status(HTTPStatus.UNAUTHORIZED).json({
```

Comment:
If an offer is not found, the error should not be `unauthorized` but `not found` instead (the famous HTTP 404)